### PR TITLE
parser: add 'INSTANT' alter algorithm

### PR DESCRIPTION
**Pull request by winkyao**
ref https://github.com/pingcap/tidb/issues/8428

Required for https://github.com/pingcap/tidb/pull/8811

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -1186,14 +1188,15 @@ AlterTableSpec:
 	{
 		$$ = &ast.AlterTableSpec{
 			Tp:    		ast.AlterTableLock,
-			LockType:   $1.(ast.LockType),
+			LockType:   	$1.(ast.LockType),
```
Comment (crazycs520) [inline parser.y]: Why change this?

File: parser.y [hunk 2]
```diff
@@ -3000,7 +3019,7 @@ TiDBKeyword:
 
 NotKeywordToken:
  "ADDDATE" | "BIT_AND" | "BIT_OR" | "BIT_XOR" | "CAST" | "COPY" | "COUNT" | "CURTIME" | "DATE_ADD" | "DATE_SUB" | "EXTRACT" | "GET_FORMAT" | "GROUP_CONCAT"
-| "INPLACE" | "INTERNAL" |"MIN" | "MAX" | "MAX_EXECUTION_TIME" | "NOW" | "RECENT" | "POSITION" | "SUBDATE" | "SUBSTRING" | "SUM"
+| "INPLACE" | "INSTANT" | "INTERNAL" |"MIN" | "MAX" | "MAX_EXECUTION_TIME" | "NOW" | "RECENT" | "POSITION" | "SUBDATE" | "SUBSTRING" | "SUM"
```
Comment (tiancaiamao) [inline parser.y]: `INSTANT` is a mysql keyword?

Comment (winkyao) [reply parser.y]: Yep.

File: parser.y (hunk 1 above)
Comment (winkyao) [reply parser.y]: go fmt change this, not me.

File: ast/ddl.go [hunk 3]
```diff
@@ -1079,6 +1079,35 @@ const (
 	LockTypeExclusive
 )
 
+// AlterAlgorithm is the algorithm of the ddl operations.
```
Comment (zimulala) [inline ast/ddl.go]: s/ddl/DDL ?

**Comment by winkyao** [comment]
@zimulala @tiancaiamao @crazycs520 PTAL

**Pull request merged**